src/index.js

```diff
@@ -54,34 +54,44 @@ function matches (node, query) {
   return (node.matches || node.msMatchesSelector).call(node, query);
 }
 
-function nodeFromHtml (html) {
-  const div = document.createElement('div');
-  div.innerHTML = html;
-  return div.firstElementChild;
+function getInstantiatedNodeWithinFixture(node, isRootNode) {
+  const isStringNode = typeof node === 'string';
+
+  // If the fixture has been removed from the document, re-insert it.
+  if (!body.contains(fixture)) {
+    body.appendChild(fixture);
+  }
+
+  if (isRootNode) {
+    setFixtureContent( node );
+  }
+
+  return isStringNode
+    ? fixture.firstElementChild
+    : node;
+}
+
+function setFixtureContent( node ) {
```

Comment:
just a nit pick.

we are using 2 times the same operation ` const isStringNode = typeof node === 'string';`

maybe we can make it more DRY like this:

```js
function getInstantiatedNodeWithinFixture(node, isRootNode) {
  const isStringNode = typeof node === 'string';

  // If the fixture has been removed from the document, re-insert it.
  if (!body.contains(fixture)) {
    body.appendChild(fixture);
  }

  if (isRootNode) {
    setFixtureContent( node, isStringNode );
  }

  return isStringNode
    ? fixture.firstElementChild
    : node;
}

function setFixtureContent( node, shouldSetChildrenViaString ) {    

    // If this is a new node, clean up the fixture.
    fixture.innerHTML = '';

    // Add the node to the fixture so it runs the connectedCallback().
    shouldSetChildrenViaString
        ? (fixture.innerHTML = node)
        : (fixture.appendChild(node));
}
```

WDYT? otherwise LGTM! thx!